format-proposal.md

```diff
@@ -0,0 +1,130 @@
+# Format proposal
+
+- **PRE: parameters and variables share their namespace**
+
+Parameters of any kind:
+
+```asp
+variable_value("a",value,1).                   % single integer
+variable_value("b",value,"1.0").               % single float
+variable_value("c",value,true).                % single bool
+variable_value("d",array,(0,value,42)).        % array of int
+variable_value("d",array,(1,value,23)).
+variable_value("e",array,(0,value,"42.0")).    % array of float
+variable_value("e",array,(1,value,"23.0")).
+variable_value("f",set,(value,17)).            % set of int
+variable_value("f",set,(range,(23,42))).       % !! INVALID FZN 
+variable_value("g",set,(value,"23.0"))).       % set of float
+variable_value("g",set,(value,"42.0"))).
+variable_value("h",array,(0,set,(value,42))).  % array of set of int
+variable_value("h",array,(0,set,(value,17))).
+variable_value("h",array,(1,set,(range,(1,5)))).
+variable_value("h",array,(1,set,(value,42))).
+variable_value("h",array,(2,set,emptyset)).
+```
+
+Variable **assignments** of any kind.
+
+```asp
+% FZN
+var int : a = 1;
+% ASP
+variable_type("a",int).
+variable_value("a",value,1).
+
+% FZN
+var float : b = 1.0;
+% ASP
+variable_type("b",float).
+variable_value("b",value,"1").
+
+% FZN
+var bool : c = true;
+% ASP
+variable_type("c",bool).
+variable_value("c",value,true).
+
+% FZN
+array [1..2] of var int : d = [42,23];
+% ASP
+variable_type("d",array(2,int)).
+variable_value("d",array,(0,value,42)).
+variable_value("d",array,(1,value,23)).
+
+% FZN
+array [1..2] of var float : e = [42.1,23.1];
+% ASP
+variable_type("e",array(2,float)).
+variable_value("e",array,(0,value,"42.1")).
+variable_value("e",array,(1,value,"23.1")).
+
+% FZN
+var set of 17..42: f = {17,23};
+% ASP
+variable_type("f",subset_of_int_range(17,42)).
+variable_value("f",set,(value,17)).
+variable_value("f",set,(value,23)).
+
+% TODO: Check if/how set of floats are allowed
+% FZN
+% var set of float: g = {23.1,42.1};
+% ASP
+% variable_value("g",set,(value,"23.1")).
+% variable_value("g",set,(value,"42.1")).
+
+% FZN
+array [1..2] of var set of 17..42: h = [{42,17},23..X];  % TODO: check empty set
+% ASP
+variable_type("h",array(2,subset_of_int_range(17,42))).
```

Comment:
Also here:
`variable_type("h",array(2,int,set,(range(17,42)))).`

Reply:
it is now
```
variable_type("h",array(2,set_of_int,(range,value,17,value,42))).
```

Reply:
I'm unsure about the additional values. Range is always two values, so there will never be something else and it will blow up the output. But your choice.

Reply:
Why will it blow up ? 
For type definitions sets and ranges always contains values (never variables) but
for ranges in assignments variables are possible which was unaccounted for previously.
Example:
```
int: x= 23;
int: y= 34;
array [1..3] of var set of 17..42: e = [17..x,x..y, y..42];
```

Reply:
Then value makes totally sense. Thx